src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -666,16 +694,32 @@ SHAMapStoreImp::health()
     }
     if (!netOPs_)
         return Health::ok;
+    assert(deleteInterval_);
 
-    constexpr static std::chrono::seconds age_threshold(60);
-    auto age = ledgerMaster_->getValidatedLedgerAge();
-    OperatingMode mode = netOPs_->getOperatingMode();
-    if (mode != OperatingMode::FULL || age > age_threshold)
+    if (healthy_)
     {
-        JLOG(journal_.warn()) << "Not deleting. state: "
-                              << app_.getOPs().strOperatingMode(mode, false)
-                              << ". age " << age.count() << 's';
-        healthy_ = false;
+        auto age = ledgerMaster_->getValidatedLedgerAge();
+        OperatingMode mode = netOPs_->getOperatingMode();
+        if (recoveryBuffer_ && mode == OperatingMode::SYNCING &&
+            age < ageThreshold_)
+        {
+            JLOG(journal_.warn())
+                << "Waiting " << recoveryBuffer_->count()
+                << "s for node to get back into sync with network. state: "
+                << app_.getOPs().strOperatingMode(mode, false) << ". age "
+                << age.count() << 's';
+            std::this_thread::sleep_for(*recoveryBuffer_);
```

Comment:
Sleeping in a call to "health" is a bit of a surprising interface. I can see why you did this. Since `health` is always called from functions on the main `run` thread, this should be OK. Still, we should probably add a comment to "health" saying the function can sleep and that it should never be called from a function outside of the `run` thread including functions that part of the public interface.